NEWS.md

```diff
@@ -3,6 +3,183 @@ XGBoost Change Log
+- Supports federated communication patterns for both CPU and GPU.
+- Supports timeout. The high-level interface parameter is currently hard-coded to 30 minutes, which we plan to improve.
+- Supports significantly more data types.
+- Supports thread-based workers.
+- Improved handling for worker errors, including better error messages when one of the peers dies during training.
+- Work with IPv6. Currently, this is only supported by the dask interface.
+- Built-in support for various operations like broadcast, allgatherV, allreduce, etc.
+
+Related PRs (#9597, #9576, #9523, #9524, #9593, #9596, #9661, #10319, #10152, #10125, #10332, #10306, #10208, #10203, #10199, #9784, #9777, #9773, #9772, #9759, #9745, #9695, #9738, #9732, #9726, #9688, #9681, #9679, #9659, #9650, #9644, #9649, #9917, #9990, #10313, #10315, #10112, #9531, #10075, #9805, #10198).
+
+The existing option of using `MPI` in RABIT is removed in the release. (#9525)
+
+### NCCL is now fetched from PyPI.
+In the previous version, XGBoost statically linked NCCL, which significantly increased the binary size and led to hitting the PyPI repository limit. However, with the new release, we have made a significant improvement. The new release can now dynamically load NCCL from an external source, reducing the binary size. For the PyPI package, the `nvidia-nccl-cu12` package will be fetched during installation. With more downstream packages reusing NCCL, we expect the user environments to be slimmer in the future as well. (#9796, #9804)
+
+### Multi-output
+We continue the work on multi-target and vector leaf in this release:
+- Revise the support for custom objectives with a new API, `XGBoosterTrainOneIter.` This new function supports strided matrices and CUDA inputs. In addition, custom objectives now return the correct shape for prediction. (#9508)
+- The `hinge` objective now supports multi-target regression (#9850)
+- Fix the gain calculation with vector leaf (#9978)
+- Support graphviz plot for multi-target tree. (#10093)
+- Fix multi-output with alternating strategies. (#9933)
+
+Please note that the feature is still in progress and not suitable for production use.
+
+### Federated Learning
+Progress has been made on federated learning with improved support for column-split, including the following updates:
+- Column split work for both CPU and GPU. In addition, categorical data is now compatible with column split. (#9562, #9609, #9611, #9628, #9539, #9578, #9685, #9623, #9613, #9511, #9384, #9595)
+-  The use of UBJson to serialize split entries for column split has been implemented, aiding vector-leaf with column-based data split. (#10059, #10055, #9702)
+- Documentation and small fixes. (#9610, #9552, #9614, #9867)
+
+### Ongoing work for SYCL support.
+
+XGBoost is developing a SYCL plugin for SYCL devices, starting with the `hist` tree method. (#10216, #9800, #10311, #9691, #10269, #10251, #10222, #10174, #10080, #10057, #10011, #10138, #10119, #10045, #9876, #9846, #9682)
```

Comment:
XGBoost is developing a SYCL plugin for SYCL devices, starting with the inference (#9800) and the `hist` tree method. (#10216, #10311, #9691, #10269, #10251, #10222, #10174, #10080, #10057, #10011, #10138, #10119, #10045, #9876, #9846, #9682)

Reply:
From a release note perspective, we can mention that XGBoost now supports launchable inference on SYCL devices, and that work on adding SYCL support for training is ongoing.

Looking ahead, I plan to complete the upstreaming of training within the next few months and then focus on improving test coverage for SYCL, particularly for Python tests.